compiler/rustc_codegen_cranelift/src/abi/mod.rs

```diff
@@ -487,7 +487,7 @@ pub(crate) fn codegen_terminator_call<'tcx>(
 
         // FIXME find a cleaner way to support varargs
         if fn_sig.c_variadic {
-            if !matches!(fn_sig.abi, Abi::C { .. }) {
+            if !fn_sig.abi.support_varargs() {
```

Comment:
This should remain the same. This is a C abi specific hack.

Reply:
Sure,  reverted it in the next commit.